### ISSUE TITLE
Poor's man multiplexing (deferred reading of replies)

### ISSUE THREAD

**Issue by antirez** [redis/redis-rb#287]
Hello redis-rb hackers!

In this feature request I propose the introduction of an optional interface, not replacing the normal interface, but only as an additional interface, that disjoins the calls to send commands and read replies.

The main rationale for this feature is to query N redis servers simultaneously without resorting to explicit multiplexing or any proper non-blocking framework.
# Feature description and assumptions

In networks where there are N Redis servers R0, R1, R2, ... Rn-1, assuming a comparable latency between a client C and any of the Redis servers, a deferred read allows to mount a pattern like the follo(r2, r3 are three Redis-rb objects

```
r1.deferred {
    reply1 = r1.set('foo','bar')
}
r2.deferred {
    reply2 = r2.set('foo','bar')
}
r3.deferred {
    reply3 = r3.set('foo',"bar')
}

reply1.read()
reply2.read()
reply3.read()
```

_Note: the interface is just an example_

The commands sent to r1, r2, r3 will leave the client about at the same time (as a TCP packet sent to the server). Because of the assumption of similar latency between C and R0, R1, R2, the replies will be available in the three sockets about at the same time, and the last two `.read()` calls will likely take minimal time to execute not paying the round time trip that will be instead payed only the first time.
# Applications

This feature will be useful every time a client needs to perform operations against multiple Redis servers in parallel. Two notable applications are:
- Client-side implementation of high availability systems where data must be written in multiple nodes.
- Optimizations of queries where data is stored across multiple nodes. For instance an interface for Redis Cluster could implement a special interface to perform queries in parallel if possible, in case data is stored in different nodes.

Thanks,
Salvatore
